docs/goals.md

```diff
@@ -0,0 +1 @@
+A prisoner's "Learning and work progress" plan contains 1 or more Goals, where a Goal consists of one or more Steps.
```

Comment:
Sparse documentation at this time! 🙄 , but this file gives us a place to continue the documentation of Goals at some point

Reply:
Is it worth adding a TODO? It will then show up in our IDEs and is more likely to be addressed.